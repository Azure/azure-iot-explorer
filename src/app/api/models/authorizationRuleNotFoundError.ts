/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import { ERROR_TYPES } from './../../constants/apiConstants';

export class AuthorizationRuleNotFoundError extends Error {
    public requiredPermissions: string[];
    constructor(requiredPermissions: string[]) {
        super(ERROR_TYPES.AUTHORIZATION_RULE_NOT_FOUND);
        this.name = this.message;
        this.requiredPermissions = requiredPermissions;
    }
}
