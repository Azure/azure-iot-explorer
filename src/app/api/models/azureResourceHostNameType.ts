/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
export enum AzureResourceHostNameType {
    IotHub = 'azure-devices',
    DeviceProvisioningService = 'azure-devices-provisioning'
}
