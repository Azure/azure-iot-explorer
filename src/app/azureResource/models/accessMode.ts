/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/

export enum AccessMode {
    resourceToken,
    connectionString
}
