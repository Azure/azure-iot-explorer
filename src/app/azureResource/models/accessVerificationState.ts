/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/

export enum AccessVerificationState {
    Verifying,
    Authorized,
    Unauthorized,
    Failed
}
