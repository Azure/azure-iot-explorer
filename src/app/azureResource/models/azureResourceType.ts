export enum AzureResourceType {
    IoTHub,
    DeviceProvisioningService
}
