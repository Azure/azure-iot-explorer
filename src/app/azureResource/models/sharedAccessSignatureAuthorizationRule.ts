/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/

export interface SharedAccessSignatureAuthorizationRule {
    accessRights: string;
    keyName: string;
    primaryKey: string;
    secondaryKey: string;
}
