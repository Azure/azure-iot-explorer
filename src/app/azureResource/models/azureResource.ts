/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/

import { AccessVerificationState } from './accessVerificationState';
import { AzureResourceIdentifier } from './azureResourceIdentifier';
import { SharedAccessSignatureAuthorizationRule } from './sharedAccessSignatureAuthorizationRule';

export interface AzureResource {
    accessVerificationState: AccessVerificationState;
    azureResourceIdentifier?: AzureResourceIdentifier;
    hostName: string;
    sharedAccessSignatureAuthorizationRules?: SharedAccessSignatureAuthorizationRule[];
    sharedAccessSignatureAuthorizationRulesLastFetched?: string;
    connectionString?: string;
}
