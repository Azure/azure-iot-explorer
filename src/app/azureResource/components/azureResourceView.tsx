/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { Route } from 'react-router-dom';
import DeviceContentContainer from '../../devices/deviceContent/components/deviceContentContainer';
import DeviceListContainer from '../../devices/deviceList/components/deviceListContainer';
import AddDeviceContainer from '../../devices/deviceList/components/addDevice/components/addDeviceContainer';
import { ROUTE_PARTS } from '../../constants/routes';
import { AccessVerificationState } from '../models/accessVerificationState';
import { useLocalizationContext } from '../../shared/contexts/localizationContext';
import { AzureResource } from '../models/azureResource';
import { ResourceKeys } from '../../../localization/resourceKeys';

export interface AzureResourceViewProps {
    activeAzureResource: AzureResource | undefined;
    currentHostName: string;
    currentUrl: string;
    setActiveAzureResourceByHostName(hostName: string): void;
}

export const AzureResourceView: React.FC<AzureResourceViewProps> = props => {
    const { activeAzureResource, currentHostName, currentUrl, setActiveAzureResourceByHostName } = props;
    const { t } = useLocalizationContext();

    React.useEffect(() => {
        if (activeAzureResource && activeAzureResource.hostName === currentHostName) {
            return;
        }

        setActiveAzureResourceByHostName(currentHostName);
    }, [currentHostName]); // tslint:disable-line:align

    if (!activeAzureResource) {
        return (<></>);
    }

    if (activeAzureResource.accessVerificationState === AccessVerificationState.Verifying) {
        return (<div>{t(ResourceKeys.azureResource.access.verifying)}</div>);
    }

    if (activeAzureResource.accessVerificationState === AccessVerificationState.Unauthorized) {
        return (<div>{t(ResourceKeys.azureResource.access.unauthorized)}</div>);
    }

    if (activeAzureResource.accessVerificationState === AccessVerificationState.Failed) {
        return (<div>{t(ResourceKeys.azureResource.access.failed)}</div>);
    }

    return (
        <>
            <Route path={`${currentUrl}/${ROUTE_PARTS.DEVICES}`} component={DeviceListContainer} exact={true}/>
            <Route path={`${currentUrl}/${ROUTE_PARTS.DEVICES}/${ROUTE_PARTS.ADD}`} component={AddDeviceContainer} exact={true} />
            <Route path={`${currentUrl}/${ROUTE_PARTS.DEVICES}/${ROUTE_PARTS.DEVICE_DETAIL}/`} component={DeviceContentContainer}/>
        </>
    );
};
