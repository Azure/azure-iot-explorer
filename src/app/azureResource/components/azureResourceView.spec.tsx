/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { shallow, mount } from 'enzyme';
import { AzureResourceView } from './azureResourceView';
import { AccessVerificationState } from '../models/accessVerificationState';

describe('AzureResourceView', () => {
    it('matches snapshot when current resource is undefined', () => {
        expect(shallow(
            <AzureResourceView
                activeAzureResource={undefined}
                currentHostName="hostName"
                currentUrl="url"
                setActiveAzureResourceByHostName={jest.fn()}
            />
        )).toMatchSnapshot();
    });

    it('matches snapshot when current resource is verifying', () => {
        expect(shallow(
            <AzureResourceView
                activeAzureResource={{
                    accessVerificationState: AccessVerificationState.Verifying,
                    hostName: 'hostName'
                }}
                currentHostName="hostName"
                currentUrl="url"
                setActiveAzureResourceByHostName={jest.fn()}
            />
        )).toMatchSnapshot();
    });

    it('matches snapshot when current resource is failed', () => {
        expect(shallow(
            <AzureResourceView
                activeAzureResource={{
                    accessVerificationState: AccessVerificationState.Failed,
                    hostName: 'hostName'
                }}
                currentHostName="hostName"
                currentUrl="url"
                setActiveAzureResourceByHostName={jest.fn()}
            />
        )).toMatchSnapshot();
    });

    it('matches snapshot when current resource is unauthorized', () => {
        expect(shallow(
            <AzureResourceView
                activeAzureResource={{
                    accessVerificationState: AccessVerificationState.Unauthorized,
                    hostName: 'hostName'
                }}
                currentHostName="hostName"
                currentUrl="url"
                setActiveAzureResourceByHostName={jest.fn()}
            />
        )).toMatchSnapshot();
    });

    it('matches snapshot when current resource is authorized', () => {
        expect(shallow(
            <AzureResourceView
                activeAzureResource={{
                    accessVerificationState: AccessVerificationState.Authorized,
                    hostName: 'hostName'
                }}
                currentHostName="hostName"
                currentUrl="url"
                setActiveAzureResourceByHostName={jest.fn()}
            />
        )).toMatchSnapshot();
    });

    it('calls setActiveAzureResourceByHostName when hostName changes', () => {
        jest.spyOn(React, 'useEffect').mockImplementation(f => f());
        const setActiveAzureResourceByHostName = jest.fn();
        const wrapper = shallow(
            <AzureResourceView
                activeAzureResource={{
                    accessVerificationState: AccessVerificationState.Unauthorized,
                    hostName: 'hostName'
                }}
                currentHostName="hostName"
                currentUrl="url"
                setActiveAzureResourceByHostName={setActiveAzureResourceByHostName}
            />
        );

        wrapper.setProps({
            activeAzureResource: {
                accessVerificationState: AccessVerificationState.Unauthorized,
                hostName: 'hostName'
            },
            currentHostName: 'newHostName',
            currentUrl: 'url',
            setActiveAzureResourceByHostName
        });

        expect(setActiveAzureResourceByHostName).toHaveBeenCalledWith('newHostName');
    });
});
