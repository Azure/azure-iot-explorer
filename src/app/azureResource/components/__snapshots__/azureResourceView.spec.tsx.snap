// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AzureResourceView matches snapshot when current resource is authorized 1`] = `
<Fragment>
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(withRouter(DeviceListComponent))",
        "type": [Function],
      }
    }
    exact={true}
    path="url/devices"
  />
  <Route
    component={[Function]}
    exact={true}
    path="url/devices/add"
  />
  <Route
    component={[Function]}
    path="url/devices/deviceDetail/"
  />
</Fragment>
`;

exports[`AzureResourceView matches snapshot when current resource is failed 1`] = `
<div>
  azureResource.access.failed
</div>
`;

exports[`AzureResourceView matches snapshot when current resource is unauthorized 1`] = `
<div>
  azureResource.access.unauthorized
</div>
`;

exports[`AzureResourceView matches snapshot when current resource is undefined 1`] = `<Fragment />`;

exports[`AzureResourceView matches snapshot when current resource is verifying 1`] = `
<div>
  azureResource.access.verifying
</div>
`;
